distro/src/avail/Avail.avail/Foundation.avail/Late Math.avail

```diff
@@ -846,3 +868,247 @@ Public stable method "_(decimal)" is
+	4304, 177, 41, 19875, 3110, 13221, 8726, 571, 7043, 6943, 1199, 352, 6435,
+	165, 1169, 3315, 978, 233, 3003, 2562, 2994, 10587, 10030, 2377, 1902, 5354,
+	4447, 1555, 263, 27027, 2283, 305, 669, 1912, 601, 6186, 429, 1930, 14873,
+	1784, 1661, 524, 3577, 236, 2360, 6146, 2850, 55637, 1753, 4178, 8466, 222,
+	2579, 2743, 2031, 2226, 2276, 374, 2132, 813, 23788, 1610, 4422, 5159, 1725,
+	3597, 3366, 14336, 579, 165, 1375, 10018, 12616, 9816, 1371, 536, 1867,
+	10864, 857, 2206, 5788, 434, 8085, 17618, 727, 3639, 1595, 4944, 2129, 2029,
+	8195, 8344, 6232, 9183, 8126, 1870, 3296, 7455, 8947, 25017, 541, 19115,
+	368, 566, 5674, 411, 522, 1027, 8215, 2050, 6544, 10049, 614, 774, 2333,
+	3007, 35201, 4706, 1152, 1785, 1028, 1540, 3743, 493, 4474, 2521, 26845,
+	8354, 864, 18915, 5465, 2447, 42, 4511, 1660, 166, 1249, 6259, 2553, 304,
+	272, 7286, 73, 6554, 899, 2816, 5197, 13330, 7054, 2818, 3199, 811, 922,
+	350, 7514, 4452, 3449, 2663, 4708, 418, 1621, 1171, 3471, 88, 11345, 412,
+	1559, 194
+>;
+
+/**
+ * Determine if an integer in {@type "[0..2^32)"} is prime. This method uses the
+ * "FJ32_256" algorithm by Forǐsek and Jančina (2015).
+ *
+ * @method "_is prime"
+ * @param "x" "whole number"
+ * @returns "boolean"
+ *          {@method "true"} if {@param "x"} is prime, {@method "false"}
+ *          otherwise.
+ * @category "Mathematics" "Integers" "Numbers"
+ * @see "http://ceur-ws.org/Vol-1326/020-Forisek.pdf"
+ */
+Public method "_is prime" is
+[
+	x : [0..2^32)
+|
+	/* This algorithm uses the limited range of input to turn the SPRP test into
+	* a deterministic test. In other approaches, this is often accomplished with
+	* multiple rounds of tests -- for [0, 2^32) this could be accomplished with
+	* a mere 3 -- however *this* approach reduces the number of SPRP checks to
+	* only 1 by first hashing the number into 256 buckets which each have a
+	* single base, defined above, proven to be correct for all hashed values. */
+	$body : boolean;
+	Assert: x > 130;
+	/* Check small divisors and ensure x is coprime with 210 */
+	If (x is even) ∨ (x mod 3 = 0) ∨ (x mod 5 = 0) ∨ (x mod 7 = 0) then
+	[
+		Exit body with false
+	];
+	hash : whole number := x;
+	/* 73244475 = 0x45d9f3b */
+	hash := ((hash >> 16) bit⊕ hash) × 73244475;
+	hash := ((hash >> 16) bit⊕ hash) × 73244475;
+	hash := ((hash >> 16) bit⊕ hash) bit∧ 255;
+	x ?→ [122..2^32) is a strong probable prime
+		to base fj32bases[hash + 1]
+];
+
+/**
+ * Determine if an integer in {@type "[0..2^64]"} is prime. This method uses a
+ * deterministic repeated Miller-Rabin test against Sinclair's (2011) 7 bases.
+ *
+ * @method "_is prime"
+ * @param "x" "whole number"
+ * @returns "boolean"
+ *          {@method "true"} if {@param "x"} is prime, {@method "false"}
+ *          otherwise.
+ * @category "Mathematics" "Integers" "Numbers"
+ * @see "http://ceur-ws.org/Vol-1326/020-Forisek.pdf"
+ */
+Public method "_is prime" is
+[
+	x : [0..2^64]
```

Comment:
With an abstract method covering [0..∞), this can simply be an override for [2^32..2^64), making the below cast into k unnecessary.